controllers/controllers.js

```diff
@@ -1,62 +1,126 @@
-const User = require("../models/User")
-const Foundation = require("../models/Foundation")
-const jwt = require("jsonwebtoken")
-const config = require("../config/index")
-const Pet = require("../models/Pet")
+const User = require("../models/User");
+const Foundation = require("../models/Foundation");
+const jwt = require("jsonwebtoken");
+const config = require("../config/index");
+const Pet = require("../models/Pet");
+const AdoptionRequest = require("../models/AdoptionRequest");
+
+const createUser = async (req, res, next) => {
+  try {
+    let newUser;
+    if (req.body.role === "user") {
+      newUser = await new User(req.body);
+      await newUser.save();
+      res.status(201).json(newUser);
+    } else if (req.body.role === "foundation") {
+      newUser = await new Foundation(req.body);
+      await newUser.save();
+      res.status(201).json(newUser);
```

Comment:
Previous comments on branch could be change this code, so you need to update after branch is merged